pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -222,13 +222,27 @@ func NewKubeGenericRuntimeManager(
 		}
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(features.ExternalRegistryCredentialProviders) && registryCredentialConfigPath != "" {
+		kubeRuntimeManager.keyring, err = credentialprovider.NewExternalProviderKeyring(registryCredentialConfigPath)
+		if err != nil {
+			klog.Errorf("Failed to read external registry credential provider config.  Kubelet may not be able to pull images: %v", err)
+			return nil, err
+		}
+	} else {
+		if utilfeature.DefaultFeatureGate.Enabled(features.ExternalRegistryCredentialProviders) && registryCredentialConfigPath == "" {
+			klog.Warningln("The feature gate ExternalRegistryCredentialProviders is enabled, but RegistryCredentialConfigPath is not set.  Using legacy credential provider.")
```

Comment:
This should be an error too.

Reply:
If the gate is enabled but the flag is not set, why should it be an error?  Isn't this the "All Alpha" case?

Reply:
Agree.

Reply:
Oh, maybe I misread. We should validate that the config isn't passed if the alpha feature is not enabled. We shouldn't warn here until we deprecate legacy providers.

Reply:
Ah right, I also read it as "you enabled this feature but did not pass the config."